### PR TITLE
Story/csi 1298 example yamls

### PR DESCRIPTION
**Pull request by igorgonibm**

### CODE REVIEW

File: deploy/kubernetes/examples/demo-statefulset-raw-block.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        volumeDevices:
+          - name: demo-volume
+            devicePath: "/dev/block"
+      volumes:
+      - name: demo-volume
+        persistentVolumeClaim:
+          claimName: demo-pvc-raw-block
+
+#      nodeSelector:
+#        kubernetes.io/hostname: NODESELECTOR
```
Comment (oriyarde) [inline deploy/kubernetes/examples/demo-statefulset-raw-block.yml]: I think `NODESELECTOR` should be something like `HOSTNAME`

**Comment by igorgonibm** [comment]
Eran approved to merge

File: deploy/kubernetes/examples/demo-statefulset-combined.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
+#############$$$$$$$$############################
```
Comment (oriyarde) [inline deploy/kubernetes/examples/demo-statefulset-combined.yaml]: maybe rename file to end with `-combined-vol-mode.yaml`? we may have more combinations in the future...

File: deploy/kubernetes/examples/template-array-secret.yaml [hunk 3]
```diff
@@ -1,16 +1,16 @@
...
 metadata:
-  name: <VALUE-1>
-  namespace: kube-system
+  name:  <NAME>
+  namespace: <NAMESPACE>
 type: Opaque
 stringData:
-  management_address: <VALUE-2,VALUE-3> # Array managment addresses
-  username: <VALUE-4>                   # Array username.
+  management_address: <ADDRESS_1,ADDRESS_2> # Array management addresses
```
Comment (oriyarde) [inline deploy/kubernetes/examples/template-array-secret.yaml]: maybe `<ADDRESS_1,ADDRESS_2>` -> `<ADDRESS_1>,<ADDRESS_2>,...`?

File: deploy/kubernetes/examples/template-storageclass.yaml [hunk 4]
```diff
@@ -1,25 +1,25 @@
...
+  csi.storage.k8s.io/provisioner-secret-name: <ARRAY_SECRET>
+  csi.storage.k8s.io/provisioner-secret-namespace: <ARRAY_SECRET_NAMESPACE>
+  csi.storage.k8s.io/controller-publish-secret-name: <ARRAY_SECRET>
+  csi.storage.k8s.io/controller-publish-secret-namespace: <ARRAY_SECRET_NAMESPACE>
 
 
-  #csi.storage.k8s.io/fstype: <VALUE_FSTYPE>   # Optional. values ext4\xfs. The default is ext4.
-  #volume_name_prefix: <VALUE_PREFIX>          # Optional.
+  #csi.storage.k8s.io/fstype: <FSTYPE>   # Optional. values ext4\xfs. The default is ext4.
+  #volume_name_prefix: <PREFIX>          # Optional
```
Comment (oriyarde) [inline deploy/kubernetes/examples/template-storageclass.yaml]: did we want to add anything about ds8k here?

**Comment by oriyarde** [comment]
@igorgonibm please open a separate ticket to fix this review after GA, and put its number here

**Comment by oriyarde** [comment]
> @igorgonibm please open a separate ticket to fix this review after GA, and put its number here

@igorgonibm still waiting for that ticket (if you think we should handle it differently, please reply).

File: deploy/kubernetes/examples/template-storageclass.yaml [hunk 5]
```diff
@@ -1,25 +1,25 @@
...
-  name: <VALUE_NAME>
+  name: <NAME>
 provisioner: block.csi.ibm.com
 parameters:
-  #capabilities:                               # Optional.
+  #capabilities:                               # Optional
   #  SpaceEfficiency=<VALUE>
   capacity:
-    pool=<VALUE_POOL_NAME>
+    pool=<POOL_NAME>
```
Comment (oriyarde) [inline deploy/kubernetes/examples/template-storageclass.yaml:15]: should this stay under `capacity:`?

File: deploy/kubernetes/examples/demo-statefulset-raw-block.yml (hunk 1 above)
Comment (oriyarde) [reply deploy/kubernetes/examples/demo-statefulset-raw-block.yml:34]: seems like it has been fixed in https://github.com/IBM/ibm-block-csi-driver/pull/167/

**Comment by oriyarde** [comment]
> > @igorgonibm please open a separate ticket to fix this review after GA, and put its number here
> 
> @igorgonibm still waiting for that ticket (if you think we should handle it differently, please reply).

@igorgonibm is it supposed to be in [CSI-1305](https://github.com/IBM/ibm-block-csi-driver/pull/167)? because I see (only) one of the conversations was resolved there...

**Pull request merged**